internals/daemon/api_warnings.go

```diff
@@ -35,41 +36,20 @@ func v1AckWarnings(c *Command, r *http.Request, _ *UserState) Response {
 	if op.Action != "okay" {
 		return BadRequest("unknown warning action %q", op.Action)
 	}
-	st := c.d.overlord.State()
-	st.Lock()
-	defer st.Unlock()
-	n := stateOkayWarnings(st, op.Timestamp)
 
-	return SyncResponse(n)
+	// Do nothing; return 0 warnings okayed.
+	return SyncResponse(0)
 }
 
 func v1GetWarnings(c *Command, r *http.Request, _ *UserState) Response {
 	query := r.URL.Query()
-	var all bool
 	sel := query.Get("select")
 	switch sel {
-	case "all":
-		all = true
-	case "pending", "":
-		all = false
+	case "all", "pending", "":
 	default:
 		return BadRequest("invalid select parameter: %q", sel)
 	}
 
-	st := c.d.overlord.State()
-	st.Lock()
-	defer st.Unlock()
-
-	var ws []*state.Warning
-	if all {
-		ws = stateAllWarnings(st)
-	} else {
-		ws, _ = statePendingWarnings(st)
-	}
-	if len(ws) == 0 {
-		// no need to confuse the issue
-		return SyncResponse([]state.Warning{})
-	}
-
-	return SyncResponse(ws)
+	// Do nothing; return empty JSON array (type doesn't matter).
+	return SyncResponse([]string{})
```

Comment:
Depending on whether we want to find out about clients using this or not (and clients finding out that they are doing an "unnecessary" API call that we might want them to stop doing eventually), a 410 status code could be worth considering here.